server/src/services/permissionCollectionsService.js

```diff
@@ -0,0 +1,71 @@
+import UserGroupsRepository from '../repositories/userGroupsRepository';
+import CollectionRepository from '../repositories/collectionRepository';
+import PermissionCollectionsRepository from '../repositories/permissionCollectionsRepository';
+import { ACCESS_GRANTED, ACCESS_DENIED, ADMIN_GROUP } from '../config/types';
+
+export const setInitialCollectionsPermissions = async (collectionId) => {
+  const groups = await UserGroupsRepository.getAll();
+  console.log(groups);
+  groups.forEach(async (group) => {
+    const permission = group.name === ADMIN_GROUP ? ACCESS_GRANTED : ACCESS_DENIED;
+    await PermissionCollectionsRepository.create({
+      userGroups_id: group.id,
+      permissionGranted: permission,
+      collections_id: collectionId,
+    });
+  });
+};
+
+export const setInitialCollectionPermissionsOnGroupAdd = async (groupId) => {
+  const collections = await CollectionRepository.getAll();
+  collections.forEach(async (collection) => {
+    const permission = ACCESS_DENIED;
+    await PermissionCollectionsRepository.create({
+      userGroups_id: groupId,
+      permissionGranted: permission,
+      collections_id: collection.id,
+    });
+  });
+};
+
+const formatResult = (data) => {
+  return data
+    .map((el) => {
+      return {
+        id: el.collections.id,
+        name: el.collections.name,
+        description: el.collections.description,
+        groups: [{ id: el.userGroups.id, name: el.userGroups.name, access: el.permissionGranted }],
+      };
+    })
+    .reduce((acc, collection) => {
+      const id = collection.id;
+      const index = acc.findIndex((el) => el.id === id);
+      if (index !== -1) {
+        acc[index].groups.push(collection.groups[0]);
+      } else {
+        acc.push(collection);
+      }
+      return acc;
+    }, []);
```

Comment:
With that implementation we fetch all rows from table and then loop through array twice what may cause some performance issues later. Let's try to simplify it a bit. As a suggestion, we can update our query to database and obtain already formatted data in response. 
It will look something like this: 
`Collections.findAll({
			attributes: ['id', 'name', 'description'],
			include: {
				associations: 'groups',
				attributes: ['id', 'name'],
				through: { attributes: ['permissionGranted'] }
			}
		});`
and in response you will have this
`{
			id: 'collectionId',
			name: 'collectionName',
			description: 'collectionDescription',
			groups: [
				{
					id: 'groupId',
					name: 'groupName',
					permissionCollections: { permissionGranted: true }
				}
			]
		}`

[here](https://sequelize.org/master/class/lib/associations/belongs-to-many.js~BelongsToMany.html) you can find more details. 
If you find better way please feel free to implement it. Hopefully you get the idea.

Reply:
done